apps/client/src/services/local-storage-dashboard/dashboard-service.ts

```diff
@@ -0,0 +1,270 @@
+  name,
+  description,
+  definition,
+}) => {
+  return new CancelablePromise<Dashboard>((resolve) => {
+    const dashboard = {
+      id: nanoid(),
+      name,
+      description,
+      definition,
+      creationDate: new Date().toISOString(),
+      lastUpdateDate: new Date().toISOString(),
+    };
+
+    const dashboards = getLocalStorageDashboards();
+    dashboards.push(dashboard);
+
+    localStorage.setItem(
+      LOCAL_STORAGE_DASHBOARDS_NAMESPACE,
+      JSON.stringify(dashboards),
+    );
+
+    resolve(dashboard);
+  });
+};
+
+export const updateDashboard: UpdateDashboard = (id, dto) => {
+  return new CancelablePromise<Dashboard>((resolve, reject) => {
+    const dashboards = getLocalStorageDashboards();
+
+    const index = dashboards.findIndex((dashboard) => dashboard.id === id);
+    if (index === -1) {
+      reject(new Error(`Dashboard with id ${id} does not exist`));
+      return;
+    }
+
+    const currentDashboard = dashboards[index];
+    if (currentDashboard == null) {
+      reject(new Error(`Dashboard with id ${id} does not exist`));
+      return;
+    }
+
+    const dashboard = {
+      ...currentDashboard,
+      ...dto,
+      lastUpdateDate: new Date().toISOString(),
+    };
+
+    dashboards[index] = dashboard;
+    localStorage.setItem(
+      LOCAL_STORAGE_DASHBOARDS_NAMESPACE,
+      JSON.stringify(dashboards),
+    );
+
+    resolve(dashboard);
+  });
+};
+
+export const deleteDashboard: DeleteDashboard = (id) => {
+  return new CancelablePromise<void>((resolve, reject) => {
+    const dashboards = getLocalStorageDashboards();
+
+    const index = dashboards.findIndex((dashboard) => dashboard.id === id);
+    if (index === -1) {
+      reject(new Error(`Dashboard with id ${id} does not exist`));
+      return;
+    }
+
+    dashboards.splice(index, 1);
+    localStorage.setItem(
+      LOCAL_STORAGE_DASHBOARDS_NAMESPACE,
+      JSON.stringify(dashboards),
+    );
+
+    resolve();
+  });
+};
+
+export const bulkDeleteDashboards: BulkDeleteDashboards = ({ ids }) => {
+  return new CancelablePromise<void>((resolve) => {
+    const dashboards = getLocalStorageDashboards();
+
+    const filteredDashboards = dashboards.filter(({ id }) => !ids.includes(id));
+
+    localStorage.setItem(
+      LOCAL_STORAGE_DASHBOARDS_NAMESPACE,
+      JSON.stringify(filteredDashboards),
+    );
+
+    resolve();
+  });
+};
+
+const getLocalStorageDashboards = () => {
+  const dashboardsSerialized = localStorage.getItem(
+    LOCAL_STORAGE_DASHBOARDS_NAMESPACE,
+  );
+
+  // TODO: dashboards type assertion
+  const dashboards: Dashboard[] = dashboardsSerialized
+    ? (JSON.parse(dashboardsSerialized) as Dashboard[])
+    : [];
+
+  return dashboards;
+};
+
+const getLocalStorageDashboard = (id: string) => {
+  const dashboards = getLocalStorageDashboards();
+
+  const dashboard = dashboards.find((dashboard) => dashboard.id === id);
+  return dashboard;
+};
+
+if (import.meta.vitest) {
```

Comment:
Is there a reason to now have a separate test file?

Reply:
nothing particular; just to simplify the module so it doesn't need to export things that has no value outside (e.g. `LOCAL_STORAGE_DASHBOARDS_NAMESPACE`)